_src/modules/data.js

```diff
@@ -0,0 +1,209 @@
+                return Array.prototype.concat.apply([], this.map(lambda));
+            },
+            writeable: false,
+            enumerable: false
+        }
+    });
+    function calculateAge(birthday) { // birthday is a date
+        var ageDifMs = Date.now() - birthday.getTime();
+        var ageDate = new Date(ageDifMs); // miliseconds from epoch
+        return Math.abs(ageDate.getUTCFullYear() - 1970);
+    }
+
+    function getDataFromSF(id, column){
+        var p = rawSF.find(d => d.id == id);
+        return p && p[column]  ? p[column] : null;
+    }
+
+    data = rawdata.flatMap((d) => {
+        let groups = d.groups
+        groups.length==0 ? groups=[11851] : groups // кто без групп? -> в группу "Не выявлено"
+        var rating=Math.floor(Math.random() * (10-4))+4
+        rating=GetRating(d.person)
+        return groups.map((b) => {
+            myGroups.add(b)
+            /*the set provide unique values of lobby groups*/
+            var fraction = isSF ? checkSFFraction(d.fraction) : d.fraction;
+            return {
+                id: d.id,
+                name: d.fullname,
+                fraction: d.fraction,
+                fraction_sf: fraction,
+                gender:d.gender,
+                age:calculateAge( new Date(d.birth_date)),
+                group: b,
+                groups: groups,
+                person: d.person,
+                rating: rating.log,
+                election_method:d.election_method,
+                committees: !isSF ? d.committees : getSFCommittee(d.committee),
+                convocations:d.convocations.length!=0 ? d.convocations.length : 1,
+                region: d.region ?  d.region.name : null,
+                goverment_body: d.goverment_body,
+                total_years: d.total_years
+            }
+        })
+    });
+    var i = 0
+    myGroups.forEach(function (value) {
+        myArrGroups.push({id: i++, val: value});
+        /*set to array*/
+    });
+}
+
+function getSFCommittee(committee){
+    if (committee === '0'
+        || committee === null
+        || committee === undefined
+        || committee === ''){
+        return ['Вне комитетов'];
+    }
+    return [committee];
+}
+
+function checkSFFraction(fraction){
+    var outOfFractionNames = ['беспартийный', 'беспартийная', 'вне партии'];
+    if (fraction === null) {
+        return 'Вне партии';
+    }
+    if (outOfFractionNames.includes(fraction.toLowerCase())){
+        return 'Вне партии';
+    }
+    return fraction;
+}
+
+function getLobbyMap(rawdata,aliases) {
+    lobby = rawdata.map((d,i) => {
+        var aliasRow = aliases.find(x=>x.name==d.name)
+        var alias = (aliasRow && aliasRow.alias!="null") ? aliasRow.alias : d.name
+        return {
+            index:i,
+            id: d.id,
+            name: d.name,
+            parent:d.parent,
+            order:d.order,
+            level:d.level,
+            tree_id:d.tree_id,
+            alias:alias
+        }
```

Comment:
так будет почище если ничего лишнего в d не передается, хотя даже если передается тоже
```suggestion
        return {
            index:i,
            alias,
            ...d,
        }
```

Reply:
Давай оставим так. В проекте нет TypeScript, а так хоть видно, что за поля пробрасываются. Тот случай, когда сокращение != упрощение.